### PR TITLE
fix ListBranches, add GetBranch

### PR DESCRIPTION
**Pull request by KshamaG**
This Pull Request has the following changes

1. Fixes the issue with ListBranches function - https://github.com/ktrysmt/go-bitbucket/issues/92
2. Adds a method GetBranch to [get a branch object within the specified repository, given the name of branch.
](https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/refs/branches/%7Bname%7D)

### CODE REVIEW

**Comment by KshamaG** [comment]
@ktrysmt Can you please review the PR?

File: bitbucket.go [hunk 1]
```diff
@@ -140,13 +140,14 @@ type RepositoryBlobOptions struct {
...
-	Owner    string `json:"owner"`
-	RepoSlug string `json:"repo_slug"`
-	Query    string `json:"q"`
-	Sort     string `json:"sort"`
-	PageNum  int    `json:"page"`
-	Pagelen  int    `json:"pagelen"`
-	MaxDepth int    `json:"max_depth"`
+	Owner      string `json:"owner"`
+	RepoSlug   string `json:"repo_slug"`
+	Query      string `json:"q"`
```
Comment (ktrysmt) [inline bitbucket.go:145]: I think it's probably better to use `query`.
```suggestion
	Query      string `json:"query"`
```

**Comment by ktrysmt** [comment]
Wow, thank you very much. I left a comment on it, please. @KshamaG

**Comment by KshamaG** [comment]
@ktrysmt Thanks for your review. The review comment has been addressed. Please check.

**Pull request merged**